### ISSUE TITLE
groupBy adds null to QueryResultType

### ISSUE THREAD

**Issue by Khartir** [phpstan/phpstan-doctrine#265]
I have the following QueryBuilder defined:
```php
/**
     * @return array<Category>
     */
    public function getAllForNavigation(): array
    {
        return $this->getEntityManager()->createQueryBuilder()
            ->select('cat')
            ->from(Category::class, 'cat')
            ->join(Navigation::class, 'nav', Join::WITH, 'nav.joinColumn = cat.joinColumn')
            ->groupBy('nav.joinColumn')
            ->getQuery()
            ->getResult();
    }
```
This reports an error:
> should return array<Category> but returns array<Category|null>
When the `groupBy`-call is removed, everything works fine.

**Comment by arnaud-lb** [comment]
Hi @Khartir 

Indeed, this should not be the case. I'm fixing this in #267.
